config/initializers/0_clowder-config.rb

```diff
@@ -29,12 +29,13 @@
 
   if kafka_security_protocol
     if kafka_security_protocol == 'sasl'
-      kafka_server_config[:security_protocol] = 'sasl_ssl'
       kafkaCaFile = Tempfile.create
       kafkaCaFile.write(first_kafka_server_config&.dig('cacert'))
       kafka_server_config[:ssl_ca_location] = kafkaCaFile.path
       kafka_server_config[:sasl_username] = first_kafka_server_config&.dig('sasl', 'username')
       kafka_server_config[:sasl_password] = first_kafka_server_config&.dig('sasl', 'password')
+      kafka_server_config[:sasl_mechanism] = first_kafka_server_config&.dig('sasl', 'saslMechanism')
+      kafka_server_config[:security_protocol] = first_kafka_server_config&.dig('sasl', 'securityProtocol')
```

Comment:
Where do we get the `saslMechanism` and the `securityProtocol` from? I'm unable to find it on the [`KafkaSASLConfig` valid keys](https://github.com/RedHatInsights/clowder-common-ruby/blob/fee91ab094f6ccd91de4eca8d63eb903e1fb90ba/lib/clowder-common-ruby/types.rb#L198-L202). They are, however, set by [Clowder Kafka provider](https://github.com/RedHatInsights/clowder/blob/bbbc4f3f6aa691278603f089af08c136b3f23289/controllers/cloud.redhat.com/config/types.go#L558-L571). I guess the ruby lib deserves this new addition.

/cc @psav

Reply:
You need to deploy in ephemeral using `--pool managed-kafka` to see these.